src/components/pagination/pagination.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import injectSheet from 'react-jss';
+import { Icon } from '../../';
+import Page from './page';
+import Stepper from './stepper';
+import Ellipsis from './ellipsis';
+import styles from './pagination-styles';
+
+const getRandomString = () =>
+  Math.random()
+    .toString(36)
+    .substr(2, 9);
+
+class Pagination extends Component {
+  constructor(props) {
+    super(props);
+    const { total, limit, selected, anchors, selectedSiblings } = this.props;
+
+    this.state = {
+      selected: parseInt(selected, 10),
+    };
+
+    this.pagesCount = Math.ceil(total / limit);
+    this.pagesToDraw = 1 + anchors * 2 + selectedSiblings * 2; // 1 = the selected item
+    this.advancedPagination = this.pagesCount > this.pagesToDraw;
+  }
+
+  getEllipsis = hiddenOnDesktop => <Ellipsis key={getRandomString()} hiddenOnDesktop={hiddenOnDesktop || !this.advancedPagination} />;
+
+  getPage(pageNumber) {
+    const { selected } = this.state;
+    return (
+      <Page
+        key={pageNumber}
+        pageNumber={pageNumber}
+        clickHandler={this.handlePageSelected}
+        isSelected={pageNumber === selected}
+        isFirst={pageNumber === 1}
+        isLast={pageNumber === this.pagesCount}
+      >
+        <span>{pageNumber}</span>
+      </Page>
+    );
+  }
+
+  handlePreviousPage = () => {
+    const { selected } = this.state;
+
+    if (selected > 1) {
+      this.handlePageSelected(selected - 1);
+    }
+  };
+
+  handleNextPage = () => {
+    const { selected } = this.state;
+
+    if (selected < this.pagesCount) {
+      this.handlePageSelected(selected + 1);
+    }
+  };
+
+  handlePageSelected = number => {
+    const { changeHandler } = this.props;
+    this.setState({
+      selected: number,
+    });
+
+    changeHandler(number);
+  };
+
+  paginate() {
```

Comment:
This is a method that from props and state render components, maybe it can be extracted to a component?